cvmfs/magic_xattr.cc

```diff
@@ -541,3 +542,22 @@ std::string UsedDirPMagicXattr::GetValue() {
 std::string VersionMagicXattr::GetValue() {
   return std::string(VERSION) + "." + std::string(CVMFS_PATCH_LEVEL);
 }
+
+std::string ExternalURLMagicXattr::GetValue() {
+  std::vector<std::string> host_chain;
+  std::vector<int> rtt;
+  unsigned current_host;
+  mount_point_->external_download_mgr()->GetHostInfo(
```

Comment:
We should still catch the case when the external download manager is `NULL`

Reply:
can you have a file marked as externalfile without an external downloadmanager? if not, then the fence should be good enough?

Reply:
Yes, the external file attribute is set in the catalog.  For it to work correctly, most likely the client needs to set an external host URL but that is nowhere enforced.